modules/apps/ai-creator/ai-creator-openai-web/src/main/java/com/liferay/ai/creator/openai/web/internal/display/context/util/AICreatorOpenAIMenuItemFactoryImpl.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.ai.creator.openai.web.internal.display.context.util;
+
+import com.liferay.ai.creator.openai.configuration.manager.AICreatorOpenAIConfigurationManager;
+import com.liferay.ai.creator.openai.display.context.util.AICreatorOpenAIMenuItemFactory;
+import com.liferay.ai.creator.openai.web.internal.constants.AICreatorOpenAIPortletKeys;
+import com.liferay.document.library.display.context.DLUIItemKeys;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.LiferayWindowState;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactory;
+import com.liferay.portal.kernel.portlet.RequestBackedPortletURLFactoryUtil;
+import com.liferay.portal.kernel.portlet.url.builder.PortletURLBuilder;
+import com.liferay.portal.kernel.servlet.taglib.ui.JavaScriptMenuItem;
+import com.liferay.portal.kernel.servlet.taglib.ui.MenuItem;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.HashMapBuilder;
+import com.liferay.portal.kernel.util.Portal;
+
+import javax.portlet.PortletMode;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Roberto Díaz
+ */
+@Component(service = AICreatorOpenAIMenuItemFactory.class)
+public class AICreatorOpenAIMenuItemFactoryImpl
+	implements AICreatorOpenAIMenuItemFactory {
+
+	@Override
+	public MenuItem createAICreatorCreateImageMenuItem(
+		ThemeDisplay themeDisplay, long folderId, long fileEntryTypeId,
+		long repositoryId) {
```

Comment:
Optional: not sure about the parameter sorting; Shouldn't `repositoryId` be put in front of folderId, and `themeDisplay` be put at the end?

Reply:
I'll do.

Once modified I'll push this to Brian.

What do you thing about https://github.com/liferay-lima/liferay-portal/pull/4919#discussion_r1399610246?

Reply:
Fixed!